src/llama.cpp

```diff
@@ -2099,6 +2103,9 @@ struct llama_hparams {
     float f_norm_eps;
     float f_norm_rms_eps;
 
+    float f_attn_logit_softcapping;
+    float f_final_logit_softcapping;
```

Comment:
Because there are already quite a few gguf circulating in the internet, can we provide a default value so that it doesn't break existing model?

Value taken from https://huggingface.co/google/gemma-2-9b/blob/main/config.json and https://huggingface.co/google/gemma-2-27b/blob/main/config.json

```cpp
    float f_attn_logit_softcapping = 50.0;
    float f_final_logit_softcapping = 30.0;
```

Then we mark the code to get KV as non-required

```cpp
        case LLM_ARCH_GEMMA2:
            {
                ml.get_key(LLM_KV_ATTENTION_LAYERNORM_RMS_EPS, hparams.f_norm_rms_eps);
                ml.get_key(LLM_KV_ATTN_LOGIT_SOFTCAPPING, hparams.f_attn_logit_softcapping, false);
                ml.get_key(LLM_KV_FINAL_LOGIT_SOFTCAPPING, hparams.f_final_logit_softcapping, false);
                hparams.attn_soft_cap = true;
```